src/Haskell/Ide/Engine/LSP/Reactor.hs

```diff
@@ -81,31 +83,34 @@ reactorSend' f = do
 
 -- ---------------------------------------------------------------------
 
+-- | Sends a single request to the scheduler so it can be be processed
+-- asynchronously.
 makeRequest :: (MonadIO m, MonadReader REnv m) => PluginRequest R -> m ()
-makeRequest req@(GReq _ _ Nothing (Just lid) _ _) = writePluginReq req lid
-makeRequest req@(IReq _ lid _ _) = writePluginReq req lid
-makeRequest req = liftIO . atomically . flip writeTChan req =<< asks reqChanIn
-
-writePluginReq :: (MonadIO m, MonadReader REnv m) => PluginRequest R -> J.LspId -> m ()
-writePluginReq req lid = do
-  wipTVar <- asks (wipReqsTVar . dispatcherEnv)
-  cin     <- asks reqChanIn
-  liftIO $ atomically $ do
-    modifyTVar wipTVar (S.insert lid)
-    writeTChan cin req
+makeRequest = Scheduler.makeRequest
+
+-- | Updates the version of a document and then sends the request to be processed
+-- asynchronously.
+updateDocumentRequest
+  :: (MonadIO m, MonadReader REnv m) => Uri -> Int -> PluginRequest R -> m ()
+updateDocumentRequest = Scheduler.updateDocumentRequest
+
+-- | Marks a s requests as cencelled by its LspId
+cancelRequest :: (MonadIO m, MonadReader REnv m) => J.LspId -> m ()
+cancelRequest lid =
+  liftIO . flip Scheduler.cancelRequest lid =<< asks scheduler
```

Comment:
This was the only function that I could not generalise, I kept getting type errors in the `Scheduler` module.